ui/components/KubeStatusIndicator.tsx

```diff
@@ -13,14 +13,16 @@ type Props = {
   suspended?: boolean;
 };
 
-export function computeReady(conditions: Condition[]): boolean {
+export function computeReady(conditions: Condition[]): string | boolean {
   const ready =
-    _.find(conditions, { type: "Ready" }) ||
-    // Deployment conditions work slightly differently;
-    // they show "Available" instead of 'Ready'
-    _.find(conditions, { type: "Available" });
-
-  return ready?.status == "True";
+    _.find(conditions, (c) => c.type === "Ready") ||
+    _.find(conditions, (c) => c.type === "Available")
```

Comment:
I find nested ternary operators terrifying. I know react programmers eat ternaries for breakfast, but I used to be a PHP programmer where ternary operators are broken, so I'm very quick to start hissing when I come across them. So I looked extra carefully, and I think I have found some potential problems.

I think that this isn't doing what the cover message says - I think `"type" == "Available", "status" == "Unknown", "reason" == "Progressing"` would make this show up as reconciling. Is that OK? Is that intentional?

I also think that the last line of the old version turned the string "True" into true, whereas the new code returns the string "True"? Is that on purpose?

The above things look like bugs to me, so I want to talk more about them before approving. The rest of this comment is offering thoughts and ideas - not instructions.

If I wanted to remove the nested ternaries, I would reorder it to use multiple returns - so
```
if (_.find(conditions, {type: "Ready", status: "Unknown", reason: "Progressing"}) {
    return "Reconciling";
}
// old function goes here
```
Some people will say multiple returns is bad code - those people are wrong. :upside_down_face: Sneaky multiple returns is bad code, but a clear pattern of "if statement, return, next if statement, return" is easy to follow - when I read the code I only have to care about the current if clause because nothing before or after will do anything, which makes the things I need to remember very small. Whereas right now, I'm having to keep the whole function in my head at once. It's not too bad in this case, but as I said I'm scared of nested ternary operators.

So then to make the old function follow the same pattern, I'd write it something like (maybe. I'm not sure this is better on its own, but I think it'd fit in better if the function grows with more if statements)
```
if (_.find(conditions, (c) => ["Ready", "Available"].includes(c.type) && c.status == "True") {
    return true;
}
return false;
```

And yes, these `_.find(` lines I've suggested are long and messy, and pulling out, like, `const isReadyish = (c) =>  ["Ready", "Available"].includes(c.type)` to use in the if conditionals would maybe be nicer.

And now that I'm in nitpick mode, "boolean or string" is a yucky return type. `ReconciliationGraph` is still treating this as a boolean, and I'm not saying it's wrong to - just, from looking at it, it's not clear if that is intended or not. Now, I like enums - enums error when you spell them wrong. Always using a string instead is OK, since that'd still force you to do `if (computeReady(parent.conditions) != "Not Ready")` and I can see it's not an accident. Another way of thinking about it is "return the valid state, if there is one" and consider "Not Ready" to not be a valid state - so, a return type of `string?` or `enum?`. I think the "True" return I complained about at the top means that you're almost there - we return a string, or false - but if that's what you're doing could you change the false to, like, null or undefined so it means "absent" rather than "untrue"?

Reply:
An alternative refactoring suggestion based on @ozamosi 's comments (enums and "early bail out" pattern):

```
export enum ReadyType {
  Reconciling = "Reconciling",
  True = "True",
}

export function computeReady(conditions: Condition[]): ReadyType | undefined {
  if (!_.find(conditions, (c) => c.type === "Ready" || c.type === "Available"))
    return undefined;

  return _.find(conditions, (c) => c.status === "Unknown") &&
    _.find(conditions, (c) => c.reason === "Progressing")
    ? ReadyType.Reconciling
    : ReadyType.True;
}
```

Now sure how it should work in more complex cases like the one Robin mentioned: `"type" == "Available", "status" == "Unknown", "reason" == "Progressing"`, so it can be modified as needed.

Other code which uses computeReady function should be updated as needed too.